x-pack/plugin/security/qa/jwt-realm/src/javaRestTest/java/org/elasticsearch/xpack/security/authc/jwt/JwtRestIT.java

```diff
@@ -508,7 +507,6 @@ public void testAuthenticationFailureIfDelegatedAuthorizationFails() throws Exce
         }
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/103308")
```

Comment:
Looks like this fails on `main` too without this change when `-Dtests.fips.enabled=true` is set. I wonder if it's because the keystore, found some information on that [here](https://github.com/elastic/elasticsearch/blob/main/TESTING.asciidoc#tls). @n1v0lg do you know if the `MutableSettingsProvider` is expected to work with fips enabled?

Reply:
ah yup good catch, I think you're right but didn't dig into the code -- we can just add:

```
assumeFalse(
    "Cannot run in FIPS mode because [...]",
    inFipsJvm()
);
```

to skip it in fips mode

Reply:
Did some more research to figure out the source of the issue. FIPS requires a password to be set on the keystore, but after adding a password I'm getting `java.lang.SecurityException: Keystore has been corrupted or tampered with` which is true, this test tampers with the keystore, so probably not something we should try to work around. Specifically [this](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java#L388) exception is thrown. Updated the test to include `assumeFalse`.

Reply:
Hmm, this [section](https://github.com/elastic/elasticsearch/blob/main/TESTING.asciidoc#keystore-password) implies that a keystore password ought to set automatically. I think what's happening in the test is that in FIPS mode, the password _is_ set but we're not passing it in the reload call so we get a reload failure. I gotta stare at this a bit longer, but we should be able to simply set a keystore password on the initial cluster config via `keystorePassword(...)` and always pass it to the reload call.

Reply:
Right, looks like it might be set based on [this](https://github.com/elastic/elasticsearch/blob/main/build-tools-internal/src/main/groovy/elasticsearch.fips.gradle#L75C1-L76C1). I tested setting it explicitly and also passing it and that works. However we still get the `Keystore has been corrupted or tampered with` after the password part is ok. Which makes sense. So I don't think there is much more to do in this case.

Reply:
I have a working fix for this but I propose to open it in a separate PR -- it'll be easier to follow that way and anyhow it's an issue that my original change introduced, unrelated to your changes. 

Let's go with skipping in FIPS (as you have it now), and I'll follow this up to bring FIPS support for the test.